stl/src/multprec.cpp

```diff
@@ -90,17 +90,19 @@ static void div(_MP_arr u,
     unsigned long long
         v0) noexcept { // divide multi-word value by scalar value (fits in lower half of unsigned long long)
     unsigned long long k = 0;
-    int ulen             = _MP_len;
-    while (0 <= --ulen) { // propagate remainder and divide
-        unsigned long long tmp = (k << shift) + u[ulen];
-        u[ulen]                = tmp / v0;
-        k                      = tmp % v0;
+    for (int ulen = _MP_len - 1; ulen >= 0; --ulen) { // propagate remainder and divide
+        unsigned long long t = (k << shift) | u[ulen];
+        u[ulen]              = t / v0;
+        k                    = t - (u[ulen] * v0); // a mod b = a - (floor(a / b) * b)
     }
 }
 
 _NODISCARD static int limit(const unsigned long long* u, int ulen) noexcept { // get index of last non-zero value
     while (u[ulen - 1] == 0) {
         --ulen;
+        // Prevent from reading outside of ulen if 0 is not found

```

Comment:
The only thing I am unsure about and that may be causing some functions to be a tiny bit slower is if this check is needed. From what I can tell it is possible for u[-1] to be read, but Analyze_assume seems to imply that is not possible, even though I cannot prove it is so.